### PR TITLE
Changes to VoeroTagsInput:

### PR DESCRIPTION
**Pull request by vahidhedayati**
Changes to VoeroTagsInput:

keyField  =  //This is the actual key you wish the selected item or what was already saved as on your db for this field
valueField = //This is the actual value field you wish to be kept in the tagsinput field box for storing on remote db from selection

remoteKey =   //This is the key field from remote or made up json array
remoteValue =  //this is the value field from remote or made up json array

disabled =  // true/false {disable input}

Please refer to  https://gist.github.com/vahidhedayati/3f27db7fb8a12f24a8a5102bae69bc06

Sorry this has been checked and file format should be left as was, it was intelij that auto formatted after a copy & paste of file content - auto indentation should only be changes now

### CODE REVIEW

**Comment by AlexMordred** [comment]
I appreciate your time, but I still don't understand what you're trying to achieve with the remote/local key/value field names, sorry. I really don't get the need for separation. The `v-model` bound data can contain both existing and non-existing values/tags, the rest should be backend work.

**Comment by vahidhedayati** [comment]
Hi Alex. 

The `remote key / value ` is to provide the end user with a useful feature of mapping what likely value the backend would be sending at any one time. 
To give an example we have two ways of looking up cities from our backend which returns city information slightly differently.
```
/city/search/y  ->  [ {id:'x', name:'y', fullName:'z'},...]
/location/search/y [ {cityId: 'x' cityName:'y', areaName:'x']
```

Whilst I appreciate what you say around back end should provide what the plugin expects to pickup, it is likely or at least in our case that backend developers don't wish to change standards of such searches to always return `[{key:'x', value:'y'}]` 

So `remoteKey` `remoteValue` simply binds to how ever I as the user of the plugin wish to capture my remote or backend data -  If i chose `areaName` as key it would always show that as the selectable value rather than `cityName` which exists on latter look up above


to `valueField` `keyField` is to allow or let the user decide on what they wish remote data that was sent as `[ {cityId: 'x' cityName:'y', areaName:'x'},..]` to be captured within the seperated json list of tags selected.
Assuming above rather than had 3 key/value fields per entry had lots more hanging off the search i.e. 

/area/search: -> `[{cityId: 'x' cityName:'y', areaName:'x', country:'a', countryId:'b', areaId:'g', locationName:'f',... },....]`


and so on  as plugin was that entire object was sent as the selected tag from remote server so what `keyField` `valueField` gave end user feature to do was to collect upon each click on a tag actual object elements they wished to capture to be saved as such.
In our case /area/searchreturned `cityId` and `cityName` but our collected tags that we wish to save onto server may actually be
```
objectToBeSaved={id:'someId', description:'a', cities:[{id:'x',name:'b'},{id:'y',name:'c'},,]
```
So we would set `keyField` as `id` where  each user selection of cityId is converted to a selected tag of `id` and where each `cityName` is captured in tag as `name` so what was sent and what is being collected to be saved. Also the change rewrote each tag to be exactly the defined `keyField` `valueField`  meaning the extra objects that hanged off the remote data for each element of `city/search` has been parsed and stripped back to the `two required key/value pairs in the tag` collected. 
In short reducing amount of unnecessary objects potentially collected per tag.

As you say it should be built into backend but sometimes backend gives what it gives, this is to give additional  flexibility.

I have recently released [vue-autocompletion](https://github.com/vahidhedayati/vue-autocompletion)  plugin which also does similar key/value pair but goes much much deeper with this concept.

some of the examples might help better understand why having remoteKey/value pair gives greater flexibility to front end developer to play with different data sets quite easily.

**Comment by AlexMordred** [comment]
I understand your problem with 2 backend endpoints now, but all of this sounds like a special case and makes the code more complicated. You can map the data on the frontend as well and then pass the data to the component via v-model as usual.
